tools/c7n_azure/c7n_azure/storage_utils.py

```diff
@@ -34,24 +33,29 @@ class StorageUtilities(object):
     def get_blob_client_by_uri(storage_uri, session=None):
         storage = StorageUtilities.get_storage_from_uri(storage_uri, session)
 
-        blob_service = BlockBlobService(account_name=storage.storage_name, account_key=storage.key)
+        blob_service = BlockBlobService(
+            account_name=storage.storage_name,
+            token_credential=storage.key)
         blob_service.create_container(storage.container_name)
         return blob_service, storage.container_name, storage.file_prefix
 
     @staticmethod
     def get_queue_client_by_uri(queue_uri, session=None):
         storage = StorageUtilities.get_storage_from_uri(queue_uri, session)
 
-        queue_service = QueueService(account_name=storage.storage_name, account_key=storage.key)
+        queue_service = QueueService(
+            account_name=storage.storage_name,
+            token_credential=storage.key)
         queue_service.create_queue(storage.container_name)
```

Comment:
In the cases where the storage account was created as part of the deployment of this policy, do we want to set the RBAC for the SP?